tinyagentos/routes/account_proxy.py

```
@@ -251,12 +267,32 @@ def _persist_join_credentials(resp: Response) -> Response:
 
 
 @router.get("/api/account/cluster/join/requests/{rid}/poll")
 async def cluster_join_poll(request: Request, rid: str):
     if not _valid_rid(rid):
         return JSONResponse({"error": "invalid request id"}, status_code=400)
     resp = await _forward_to(request, "GET", f"{_JOIN_BASE}/requests/{rid}/poll")
-    return _persist_join_credentials(resp)
+    out, join_intent = _persist_join_credentials(resp)
+    if join_intent and not await mesh.is_joined():
+        # Fire-and-forget so the slow `tailscale up` never blocks the poll.
+        asyncio.create_task(_run_mesh_join(join_intent))
```

Ruff (0.15.20)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)